packages/dosespot-react/src/useDoseSpotClinicFormulary.ts

```diff
@@ -1,54 +1,72 @@
 // SPDX-FileCopyrightText: Copyright Orangebot, Inc. and Medplum contributors
 // SPDX-License-Identifier: Apache-2.0
-import { MedicationKnowledge } from '@medplum/fhirtypes';
+import { isCodeableConcept } from '@medplum/core';
+import { CodeableConcept, Coding, MedicationKnowledge } from '@medplum/fhirtypes';
 import { useMedplum } from '@medplum/react-hooks';
 import { useCallback, useState } from 'react';
 import { DOSESPOT_ADD_FAVORITE_MEDICATION_BOT, DOSESPOT_SEARCH_MEDICATIONS_BOT } from './common';
 
 export interface DoseSpotClinicFormularyReturn {
   state: DoseSpotClinicFormularyState;
   /**
-   * Search for DoseSpot Medications and returns array of synthetic MedicationKnowledge objects that are not yet saved to the FHIR server
+   * Search for DoseSpot Medications and returns array of temporary MedicationKnowledge objects that are not yet saved to the FHIR server
    */
-  readonly searchMedications: (searchTerm: string) => Promise<MedicationKnowledge[]>;
+  readonly searchMedications: (searchTerm: string) => Promise<CodeableConcept[]>;
   /**
    * Set the currently selected medication
    */
-  readonly setSelectedMedication: (medication: MedicationKnowledge | undefined) => void;
+  readonly setSelectedMedication: (medication: CodeableConcept | Coding | undefined) => void;
   /**
    * Set the directions for the currently selected medication
    */
-  readonly setDirections: (directions: string | undefined) => void;
+  readonly setSelectedMedicationDirections: (directions: string | undefined) => void;
   /**
-   * Add a DoseSpot Medication to the Clinic's favorites and returns the MedicationKnowledge object that was added
+   * Save a DoseSpot Medication to the Clinic's favorites and returns the MedicationKnowledge object that was saved
    */
-  readonly addFavoriteMedication: () => Promise<MedicationKnowledge>;
+  readonly saveFavoriteMedication: () => Promise<MedicationKnowledge>;
   /**
-   * Helper function to get the name of a medication
+   * Clear the state
    */
-  readonly getMedicationName: (medication: MedicationKnowledge | undefined) => string;
+  readonly clear: () => void;
 }
 
 export interface DoseSpotClinicFormularyState {
-  selectedMedication: MedicationKnowledge | undefined;
+  selectedMedication: CodeableConcept | Coding | undefined;
   directions: string | undefined;
 }
 
 export function useDoseSpotClinicFormulary(): DoseSpotClinicFormularyReturn {
   const [directions, privateSetDirections] = useState<string | undefined>(undefined);
-  const [selectedMedication, privateSetSelectedMedication] = useState<MedicationKnowledge | undefined>(undefined);
+  const [selectedMedication, privateSetSelectedMedication] = useState<CodeableConcept | Coding | undefined>(undefined);
```

Comment:
nit: to simplify the state for the end user, we can make this `CodeableConcept |  undefined`

We can accept `CodeableConcept | Coding | undefined`, but if it's a coding we can just wrap it in a `CodeableConcept`. 

That way the client doesn't have to do many checks to see if we've saved a `Coding` or `CodeableConcept` on their behalf